### PR TITLE
chore: added clippy to github ci

### PR DESCRIPTION
**Pull request by ZigTag**
I added the workflow as mentioned in discord

Discord: Falcon#1379

### CODE REVIEW

File: .github/workflows/main-ci.yml [hunk 1]
```diff
@@ -12,11 +12,15 @@ jobs:
...
           override: true
-          components: rustfmt
+          components: rustfmt, clippy
       - uses: actions-rs/cargo@v1
         with:
           command: fmt
           args: --all -- --check
+      - uses: actions-rs/clippy-check@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (TheButlah) [inline .github/workflows/main-ci.yml]: whats the token for?

Comment (ZigTag) [reply .github/workflows/main-ci.yml]: It adds comments onto the PR directly and it needs the token for that.
ref: [link](https://github.com/actions-rs/clippy-check#limitations)

Comment (TheButlah) [reply .github/workflows/main-ci.yml]: I'm not sure it actually needs it though, it looks like `clippy-check` has not been developed in a long time and we can just use the `cargo` action:
https://github.com/actions-rs/meta/blob/master/recipes/quickstart.md

The existing CI manages to show the warnings in the PR without any such token

Comment (ZigTag) [reply .github/workflows/main-ci.yml]: Ah your right, had not noticed that. Will change

File: .github/workflows/main-ci.yml [hunk 2]
```diff
@@ -12,11 +12,15 @@ jobs:
           profile: minimal
           toolchain: nightly
           override: true
-          components: rustfmt
+          components: rustfmt, clippy
       - uses: actions-rs/cargo@v1
         with:
           command: fmt
           args: --all -- --check
```
Comment (TheButlah) [inline .github/workflows/main-ci.yml:24]: `cargo clippy` makes `cargo check` redundant. Remove `cargo check`

```suggestion
```

**Review by TheButlah — CHANGES_REQUESTED** [review]
~~I don't know how github editor works~~~

**Review by TheButlah — APPROVED** [review]

**Pull request merged**